app/controllers/shopping_carts_controller.rb

```diff
@@ -2,11 +2,11 @@ class ShoppingCartsController < ApplicationController
   before_action :get_shopping_cart
 
   def checkout
-
+    @cart = ShoppingCart.find(session[:cart_id])
```

Comment:
You don't need to do this. `@cart` is being set by the `before_action`  `:get_shopping_cart`